nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py

```diff
@@ -149,322 +170,321 @@ def create_site_location_type_locations(
         for prefix in prefixes:
-            prefix.location = Location.objects.get(name=prefix.site.name, location_type=site_lt)
+            prefix.location = prefix.site.migrated_location
         Prefix.objects.bulk_update(prefixes, ["location"], 1000)
 
         vlangroups = VLANGroup.objects.filter(location__isnull=True, site__isnull=False).select_related("site")
         for vlangroup in vlangroups:
-            vlangroup.location = Location.objects.get(name=vlangroup.site.name, location_type=site_lt)
+            vlangroup.location = vlangroup.site.migrated_location
         VLANGroup.objects.bulk_update(vlangroups, ["location"], 1000)
 
         vlans = VLAN.objects.filter(location__isnull=True, site__isnull=False).select_related("site")
         for vlan in vlans:
-            vlan.location = Location.objects.get(name=vlan.site.name, location_type=site_lt)
+            vlan.location = vlan.site.migrated_location
         VLAN.objects.bulk_update(vlans, ["location"], 1000)
 
         # Virtualization App
         clusters = Cluster.objects.filter(location__isnull=True, site__isnull=False).select_related("site")
         for cluster in clusters:
-            cluster.location = Location.objects.get(name=cluster.site.name, location_type=site_lt)
+            cluster.location = cluster.site.migrated_location
         Cluster.objects.bulk_update(clusters, ["location"], 1000)
 
+    # Region and Site Related models
+    ComputedField = apps.get_model("extras", "computedfield")
+    ConfigContext = apps.get_model("extras", "configcontext")
+    CustomField = apps.get_model("extras", "customfield")
+    DynamicGroup = apps.get_model("extras", "DynamicGroup")
+    ExportTemplate = apps.get_model("extras", "exporttemplate")
+    JobHook = apps.get_model("extras", "jobhook")
+    Note = apps.get_model("extras", "note")
+    ObjectChange = apps.get_model("extras", "objectchange")
+    Relationship = apps.get_model("extras", "relationship")
+    RelationshipAssociation = apps.get_model("extras", "relationshipassociation")
+    WebHook = apps.get_model("extras", "webhook")
+
+    computed_fields = ComputedField.objects.filter(content_type=model_ct)
+    for cf in computed_fields:
+        cf.content_type = location_ct
+    ComputedField.objects.bulk_update(computed_fields, ["content_type"], 1000)
+
+    if model == "region":
+        ccs = ConfigContext.objects.filter(regions__isnull=False).prefetch_related("locations", f"{model}s")
+    else:
+        ccs = ConfigContext.objects.filter(sites__isnull=False).prefetch_related("locations", f"{model}s")
+    for cc in ccs:
+        if model == "region":
+            model_name_list = list(cc.regions.all().values_list("name", flat=True))
+        else:
+            model_name_list = list(cc.sites.all().values_list("name", flat=True))
+
+        model_locs = list(Location.objects.filter(name__in=model_name_list, location_type=model_lt))
```

Comment:
Seems like maybe using `values_list("pk"...)` and `filter(pk__in...)` might be marginally more efficient?